hc2mqtt

```diff
@@ -78,65 +124,188 @@ def client_connect(client, device, mqtt_topic):
 				raise Exception(f"Payload {msg} is not correctly formatted")
 		except Exception as e:
 			print("ERROR", e, file=sys.stderr)
+	
+	def on_specific_message(client, userdata, msg):
+		global dev
+		mqtt_state = msg.payload.decode()
+		topic = msg.topic
+		if debugVerbose: print(f"debug topic: {topic}")
+		
+		#extract feature name from mqttTopic/feature_name/value/set
+		prefixLength = len(mqtt_topic) + 1 # +1 cause of backslash
+		suffixLength = -10 # len("/value/set")
+		node_name = topic[prefixLength:suffixLength]
+		if debugVerbose: print(now(),f"received mqtt specific message '{mqtt_state}' on topic '{topic}' extracted node_name '{node_name}'")
+		
+		try:	
+			msg = {}
+			if (features[node_name][0] == "BSH.Common.Root.ActiveProgram"):
+				print(programs)
+				print(programs[mqtt_state])
+				msg["program"] = programs[mqtt_state][1]
+				#Maybe* TODO possible options according to this json format (thanks to @chris-mc1): [{"program":8196, "options":[{"uid":558,"value":0},{"uid":5123,"value":false},{"uid":5126,"value":false},{"uid":5127,"value":false}]}]
+				#*Not easy to implement together with Homie-MQTT protocol (only one value per MQTT topic). Therefore just set the options before starting the porgram.
+				dev.post("/ro/activeProgram",data=msg)
+			else:		
+				#default settable value
+				msg["uid"] = int(features[node_name][1])
+				msg["value"] = mqtt_state
+				dev.post("/ro/values",data=msg)
+			if debugVerbose: print(f"debug msg: {msg}")
+		except Exception as e:
+			print("ERROR on specific message", e, file=sys.stderr)
+	
+	def publish_single(node_name, value):
+		if (node_name in state):
+			if (state[node_name] != value):
+				#to reduce MQTT network traffic spamming only if it is a new value
+				publish_general("/" + node_name + "/value", value)
+		else:
+			if debugVerbose: print(f"node_name {node_name} not in state - not published")
+	
+	def publish_general(topic_sub_name, value):
+		channel = mqtt_topic + topic_sub_name
+		if debugVerbose: print(now(), device["name"], f"publish '{value}' to '{channel}'")
+		client.publish(channel,value,qos=1,retain=True)
```

Comment:
dishwasher tryes to publish '{'length': 0, 'list': []}' to 'homie/dishwasher/Program/value'
--> payload must be a string, bytearray, int, float or None.
--> quick fix 
 if isinstance(value, (str, bytearray, int, float)):

Reply:
Thanks. I didn't noticed that this happens, but will just include your isinstance check to be safe for future ;)